src/main/kotlin/TelegramBotService.kt

```diff
@@ -9,105 +13,175 @@ import java.nio.charset.StandardCharsets
+const val STATISTICS_BUTTON: String = "statistics_clicked"
+const val BACK_TO_MENU_BUTTON: String = "back_to_menu_clicked"
 
 const val CALLBACK_DATA_ANSWER_PREFIX = "answer_"
 
-data class UpdateData(
-    val updateId: String,
-    val chatId: String,
+@Serializable
+data class Update(
+    @SerialName("update_id")
+    val updateId: Long,
+    @SerialName("message")
+    val message: Message? = null,
+    @SerialName("callback_query")
+    val callbackQuery: CallbackQuery? = null,
+)
+
+@Serializable
+data class Message(
+    @SerialName("text")
+    val text: String,
+    @SerialName("chat")
+    val chat: Chat,
+)
+
+@Serializable
+data class CallbackQuery(
+    @SerialName("data")
+    val data: String? = null,
+    @SerialName("message")
+    val message: Message? = null,
+)
+
+@Serializable
+data class Chat(
+    @SerialName("id")
+    val id: Long,
+)
+
+@Serializable
+data class Response(
+    @SerialName("result")
+    val result: List<Update>,
+)
+
+@Serializable
+data class SendMessageRequest(
+    @SerialName("chat_id")
+    val chatId: Long,
+    @SerialName("text")
+    val text: String,
+    @SerialName("reply_markup")
+    val replyMarkup: ReplyMarkup,
+)
+
+@Serializable
+data class ReplyMarkup(
+    @SerialName("inline_keyboard")
+    val inlineKeyboard: List<List<InlineKeyboard>>,
+)
+
+@Serializable
+data class InlineKeyboard(
+    @SerialName("callback_data")
+    val callbackData: String,
+    @SerialName("text")
     val text: String,
-    val data: String,
+)
+
+data class UpdateData(
+    val updateId: Long,
+    val chatId: Long,
+    val message: String,
+    val callbackData: String,
 ) {
 
     override fun toString(): String {
         return """
             $ANSI_GREEN
             <-|UpdateId = [$updateId]
             <-|ChatId = [$chatId]
-            <-|Text = [$text]
-            <-|Data = [$data]
+            <-|Message = [$message]
+            <-|CallbackData = [$callbackData]
             $ANSI_RESET
         """.trimMargin("<-|")
     }
 }
 
 data class TelegramBotService(
     private val botToken: String,
-    private var lastUpdateId: Int = 0,
+    private var lastUpdateId: Long = 0L,
 ) {
+
+    private val json = Json {
+        ignoreUnknownKeys = true
+    }
+
     // Функция получения обновления и данных если они пришли с ответом.
     fun getUpdates(): UpdateData? {
         val urlGetUpdates = "$BASE_URL/bot$botToken/getUpdates?offset=$lastUpdateId"
-        val update: String = getResponse(urlGetUpdates).body()
-        println("Response from getUpdates: $update")
+        val responseString: String = getResponse(urlGetUpdates).body()
+        println("Response from getUpdates: $responseString")
 
-        val newUpdateId = getDataFromUpdate("\"update_id\":(\\d+)", update)?.toIntOrNull()
+        val response: Response = json.decodeFromString(responseString)
+        val update = response.result
+        val firstUpdate = update.firstOrNull()
+        val updateId = firstUpdate?.updateId
```

Comment:
можно немного сократить код и убрать ниже if, если заменим на такую конструкцию         
```suggestion
        val updateId = firstUpdate?.updateId ?: return null
```